Source/Core/CoplanarPolygonGeometryLibrary.js

```diff
@@ -0,0 +1,131 @@
+define([
+        './defined',
+        './Cartesian2',
+        './Cartesian3',
+        './Check',
+        './IntersectionTests',
+        './Math',
+        './Matrix3',
+        './OrientedBoundingBox',
+        './Plane',
+        './Ray'
+    ], function(
+        defined,
+        Cartesian2,
+        Cartesian3,
+        Check,
+        IntersectionTests,
+        CesiumMath,
+        Matrix3,
+        OrientedBoundingBox,
+        Plane,
+        Ray
+    ) {
+    'use strict';
+
+    /**
+     * @private
+     */
+    var PolygonGeometryLibrary = {};
+
+    var scratchIntersectionPoint = new Cartesian3();
+    var scratchXAxis = new Cartesian3();
+    var scratchYAxis = new Cartesian3();
+    var scratchZAxis = new Cartesian3();
+    var scratchOrigin = new Cartesian3();
+    var scratchNormal = new Cartesian3();
+    var scratchRay = new Ray();
+    var scratchPlane = new Plane(Cartesian3.UNIT_X, 0);
+    var obbScratch = new OrientedBoundingBox();
+
+    // call after removeDuplicates
+    PolygonGeometryLibrary.projectTo2D = function(positions, positionsResult, normalResult, tangentResult, bitangentResult) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('positions', positions);
+        Check.defined('positionsResult', positionsResult);
+        //>>includeEnd('debug');
+
+        var obb = OrientedBoundingBox.fromPoints(positions, obbScratch);
+        var halfAxes = obb.halfAxes;
+        var xAxis = Matrix3.getColumn(halfAxes, 0, scratchXAxis);
+        var yAxis = Matrix3.getColumn(halfAxes, 1, scratchYAxis);
+        var zAxis = Matrix3.getColumn(halfAxes, 2, scratchZAxis);
+
+        var xMag = Cartesian3.magnitude(xAxis);
+        var yMag = Cartesian3.magnitude(yAxis);
+        var zMag = Cartesian3.magnitude(zAxis);
+        var min = Math.min(xMag, yMag, zMag);
+
+        var i;
+        // If all the points are on a line, just remove one of the zero dimensions
+        if ((xMag === 0 && (yMag === 0 || zMag === 0)) || (yMag === 0 && zMag === 0)) {
+            return;
+        }
+        var center = obb.center;
+        var planeXAxis;
+        var planeYAxis;
+        var origin = Cartesian3.clone(center, scratchOrigin);
+        var normal;
+        if (min === xMag) {
+            if (min !== 0) {
+                origin = Cartesian3.add(origin, xAxis, origin);
```

Comment:
Why move the origin from the bounding box center?

Reply:
It's because all the points are already in a plane and I need a new plane to project them onto =P
There's definitely a better way to do this.  I hadn't given this logic any thought since I wrote it at our first code sprint down the shore.  Thanks for pointing that out!

Reply:
Oh no wait!  It's because I want the intersection plane to be where the OBB box face is instead of at the center of the box.